File: angularapp/src/app/services/feedback.service.ts
import { Injectable } from '@angular/core';
import { Feedback } from '../models/feedback.model';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { FeedbackDto } from '../models/feedback-dto.model';

@Injectable({
  providedIn: 'root'
})
export class FeedbackService {


  private apiUrl: string = "https://8080-eacadeababedeffaddfeeacfcadcfdab.premiumproject.examly.io/api/feedback";


  constructor(private http: HttpClient) { }

  sendFeedback(feedbackDto: FeedbackDto): Observable<Feedback>{
    return this.http.post<Feedback>(this.apiUrl, feedbackDto);
  }
 
  getAllFeedbacksByUserId(userId: number): Observable<Feedback[]>{
    return this.http.get<Feedback[]>(`${this.apiUrl}/user/${userId}`);
  }
 
  deleteFeedback(feedbackId: number): Observable<void>{
    return this.http.delete<void>(`${this.apiUrl}/${feedbackId}`);
  }
 
  getFeedbacks(): Observable<Feedback[]>{
    return this.http.get<Feedback[]>(this.apiUrl);
  }
}

File: springapp/src/main/java/com/examly/springapp/service/FeedbackService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.model.Feedback;
import com.examly.springapp.model.FeedbackDTO;

import jakarta.persistence.EntityNotFoundException;

public interface FeedbackService {

    Feedback createFeedback(FeedbackDTO feedbackDto) throws Exception;

    Feedback getFeedbackById(Long feedbackId) throws EntityNotFoundException;

    List<Feedback> getAllFeedbacks();

    String deleteFeedback(Long feedbackId) throws EntityNotFoundException;

    List<Feedback> getFeedbacksByUserId(Long userId) throws EntityNotFoundException;

}
